### PR TITLE
This add sre user group to priviledged scc provider

### PR DESCRIPTION
**Pull request by wanghaoran1988**

### CODE REVIEW

**Comment by cblecker** [comment]
Why would we try and patch the existing, as opposed to creating a SRE specific SCC that we can authoritatively manage? That way the can’t can be reversed/changed, as opposed to a one way patch operation.

**Comment by wanghaoran1988** [comment]
> Why would we try and patch the existing, as opposed to creating a SRE specific SCC that we can authoritatively manage? That way the can’t can be reversed/changed, as opposed to a one way patch operation.

Good suggestion !

**Comment by wanghaoran1988** [comment]
Updated

File: deploy/osd-scc/01-osd-sre-scc.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+apiVersion: security.openshift.io/v1
+kind: SecurityContextConstraints
+metadata:
```
Comment (cblecker) [inline deploy/osd-scc/01-osd-sre-scc.yaml]: I think copying the description from the default `securitycontextconstraints.security.openshift.io/privileged` object is a good idea:
```suggestion
metadata:
  annotations:
    kubernetes.io/description: 'privileged allows access to all privileged and host
      features and the ability to run as any user, any group, any fsGroup, and with
      any SELinux context.  WARNING: this is the most relaxed SCC and should be used
      only for cluster administration. Grant with caution.'
```

**Comment by cblecker** [comment]
one comment, otherwise lgtm

**Comment by wanghaoran1988** [comment]
@cblecker updated

**Comment by cblecker** [comment]
Oh, one other comment. This can probably go in the `deploy/sre-authorization` folder as it all relates to authorization for the sre group, rather than it's own folder.

**Comment by wanghaoran1988** [comment]
> Oh, one other comment. This can probably go in the `deploy/sre-authorization` folder as it all relates to authorization for the sre group, rather than it's own folder.

That make sense, updated.

**Comment by wanghaoran1988** [comment]
@aweiteka Would you mind take a look?

**Comment by jewzaam** [comment]
/hold

Why are we making this change?  Technically looks reasonable, but I'm just not clear on why we need this SCC setup for SRE.  What day to day support activities require this?

**Comment by wanghaoran1988** [comment]
oc debug node will run a privileged pod that need this.

**Comment by jewzaam** [comment]
I'm able to run oc debug on a cluster I have setup without this change.  Is there something specific we can't do without this in place?

```
$ oc whoami
nmalik@redhat.com

$ oc get SecurityContextConstraints
NAME               AGE
anyuid             25h
hostaccess         25h
hostmount-anyuid   25h
hostnetwork        25h
node-exporter      25h
nonroot            25h
privileged         25h
restricted         25h

$ oc debug node/ip-10-0-134-2.ec2.internal
Starting pod/ip-10-0-134-2ec2internal-debug ...
To use host binaries, run `chroot /host`
If you don't see a command prompt, try pressing enter.
sh-4.2# chroot /host
sh-4.4# curl http://169.254.169.254/latest/meta-data/instance-type
m5.xlarge
sh-4.4# exit
sh-4.2# exit

Removing debug pod ...
```

**Comment by wanghaoran1988** [comment]
@jewzaam I met problem on v4-stg and v4-prod cluster with errors:
```
Error from server (Forbidden): pods "ip-10-0-128-189us-east-2computeinternal-debug" is forbidden: unable to validate against any security context constraint: [provider restricted: .spec.securityContext.hostNetwork: Invalid value: true: Host network is not allowed to be used provider restricted: .spec.securityContext.hostPID: Invalid value: true: Host PID is not allowed to be used spec.volumes[0]: Invalid value: "hostPath": hostPath volumes are not allowed to be used spec.containers[0].securityContext.securityContext.runAsUser: Invalid value: 0: must be in the ranges: [1000420000, 1000429999] spec.containers[0].securityContext.privileged: Invalid value: true: Privileged containers are not allowed spec.containers[0].securityContext.hostNetwork: Invalid value: true: Host network is not allowed to be used spec.containers[0].securityContext.hostPID: Invalid value: true: Host PID is not allowed to be used]
```

**Comment by jewzaam** [comment]
@wanghaoran1988 in prod I see your user is in the `osd-sre-admins` group on cluster `osd-v4prod-aws`.  I am not granted any special privilege directly to my user.  I am only a member of `osd-sre-admins` group, same as your user.  I am able to debug the node you mention.

```
$ oc whoami
nmalik
$ oc debug node/ip-10-0-128-189.us-east-2.compute.internal
Starting pod/ip-10-0-128-189us-east-2computeinternal-debug ...
To use host binaries, run `chroot /host`
If you don't see a command prompt, try pressing enter.
sh-4.2# chroot /host
sh-4.4# cat /etc/redhat-release 
Red Hat Enterprise Linux CoreOS release 4.1
sh-4.4# exit
sh-4.2# exit

Removing debug pod ...
```

Could you please provide more info, such as your user name and what exact command you're executing to get this error?

**Comment by wanghaoran1988** [comment]
@jewzaam here is the infor needed
```
$oc debug node/ip-10-0-128-189.us-east-2.compute.internal                                     [8:18:37]
Error from server (Forbidden): pods "ip-10-0-128-189us-east-2computeinternal-debug" is forbidden: unable to validate against any security context constraint: [provider restricted: .spec.securityContext.hostNetwork: Invalid value: true: Host network is not allowed to be used provider restricted: .spec.securityContext.hostPID: Invalid value: true: Host PID is not allowed to be used spec.volumes[0]: Invalid value: "hostPath": hostPath volumes are not allowed to be used spec.containers[0].securityContext.securityContext.runAsUser: Invalid value: 0: must be in the ranges: [1000420000, 1000429999] spec.containers[0].securityContext.privileged: Invalid value: true: Privileged containers are not allowed spec.containers[0].securityContext.hostNetwork: Invalid value: true: Host network is not allowed to be used spec.containers[0].securityContext.hostPID: Invalid value: true: Host PID is not allowed to be used]

 $ oc version                                                                                   [8:18:44]
Client Version: version.Info{Major:"4", Minor:"1+", GitVersion:"v4.1.8", GitCommit:"4d531fa7e", GitTreeState:"clean", BuildDate:"2019-07-21T12:18:11Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"13+", GitVersion:"v1.13.4+3a25c9b", GitCommit:"3a25c9b", GitTreeState:"clean", BuildDate:"2019-07-18T00:10:31Z", GoVersion:"go1.11.6", Compiler:"gc", Platform:"linux/amd64"}

$ oc whoami                                                                                    [8:18:51]
haowang

```

**Comment by jewzaam** [comment]
I didn't find 4.1.8 cli but tried w/ 4.1.9

```
$ ./oc version
Client Version: version.Info{Major:"4", Minor:"1+", GitVersion:"v4.1.9", GitCommit:"0cb23916f", GitTreeState:"clean", BuildDate:"2019-07-28T12:21:40Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"13+", GitVersion:"v1.13.4+3a25c9b", GitCommit:"3a25c9b", GitTreeState:"clean", BuildDate:"2019-07-18T00:10:31Z", GoVersion:"go1.11.6", Compiler:"gc", Platform:"linux/amd64"}
$ ./oc whoami
nmalik@redhat.com
$ ./oc debug node/ip-10-0-128-189.us-east-2.compute.internal
Starting pod/ip-10-0-128-189us-east-2computeinternal-debug ...
To use host binaries, run `chroot /host`
If you don't see a command prompt, try pressing enter.
sh-4.2# exit

Removing debug pod ...
```

Maybe is something wrong with your user on this cluster.  I've deleted it so it will be recreated when you log in again.  Could you try again and let me know?  I tried w/ --as and it works too:

```
$ ./oc whoami
kube:admin
$ ./oc --as-group=osd-sre-admins --as=haowang debug node/ip-10-0-128-189.us-east-2.compute.internal
Starting pod/ip-10-0-128-189us-east-2computeinternal-debug ...
To use host binaries, run `chroot /host`
If you don't see a command prompt, try pressing enter.
sh-4.2# exit

Removing debug pod ...
```

**Comment by wanghaoran1988** [comment]
@jewzaam I know the problem now, this is not related to user account, this is related which project currently you are in when you run the debug command, could you try this? you will met the same probem I think.
1. switch to another project
   #oc project openshift-monitoring
2. then run the oc debug

Let me know if you met the same errors

**Comment by cblecker** [comment]
@wanghaoran1988 I was able to confirm your findings!
- If my namespace/project is *not* `default`, then I get the SCC error.
- If I use the `-n default` flag to force the `default` namespace/project, then I can `oc debug` successfully.

**Comment by jewzaam** [comment]
Same, reproduced in a different namespace.  Recommend we simply update SOPs to indicate running oc debug in default.  I've updated what we have tracked so far.  Worth sending a note to the team more broadly as well.

**Comment by cblecker** [comment]
Updated our SOP doc. I think we can close this.

**Pull request closed without merging**